python_modules/dagster/dagster/_core/execution/plan/execute_step.py

```diff
@@ -638,7 +638,8 @@ def _get_output_asset_events(
                     **all_unpartitioned_asset_metadata,
                     **(partition_scoped_metadata or {}),
                 }
-                all_tags.update(
+                tags_for_event = {**all_tags}
```

Comment:
should probably add a comment here -- it makes sense when you see it but adding a bit of extra attention to this detail can't hurt